ADAL/src/telemetry/ADTelemetryAPIEvent.m

```diff
@@ -162,29 +189,50 @@ - (void)addAggregatedPropertiesToDictionary:(NSMutableDictionary *)eventToBeDisp
 {
     [super addAggregatedPropertiesToDictionary:eventToBeDispatched];
     
-    NSArray* properties = [self getProperties];
-    for (ADTelemetryProperty* property in properties)
+    NSDictionary* properties = [self getProperties];
+    for (NSString* name in properties)
     {
-        if ([property.name isEqualToString:AD_TELEMETRY_AUTHORITY_TYPE]
-            ||[property.name isEqualToString:AD_TELEMETRY_AUTHORITY_VALIDATION_STATUS]
-            ||[property.name isEqualToString:AD_TELEMETRY_EXTENDED_EXPIRES_ON_SETTING]
-            ||[property.name isEqualToString:AD_TELEMETRY_PROMPT_BEHAVIOR]
-            ||[property.name isEqualToString:AD_TELEMETRY_RESULT_STATUS]
-            ||[property.name isEqualToString:AD_TELEMETRY_IDP]
-            ||[property.name isEqualToString:AD_TELEMETRY_TENANT_ID]
-            ||[property.name isEqualToString:AD_TELEMETRY_USER_ID]
-            ||[property.name isEqualToString:AD_TELEMETRY_RESPONSE_TIME]
-            ||[property.name isEqualToString:AD_TELEMETRY_CLIENT_ID]
-            ||[property.name isEqualToString:AD_TELEMETRY_API_ID]
-            ||[property.name isEqualToString:AD_TELEMETRY_USER_CANCEL]
-            ||[property.name isEqualToString:AD_TELEMETRY_ERROR_CODE]
-            ||[property.name isEqualToString:AD_TELEMETRY_ERROR_DOMAIN]
-            ||[property.name isEqualToString:AD_TELEMETRY_PROTOCOL_CODE]
-            ||[property.name isEqualToString:AD_TELEMETRY_ERROR_DESCRIPTION])
+        if ([name isEqualToString:AD_TELEMETRY_AUTHORITY_TYPE]
+            ||[name isEqualToString:AD_TELEMETRY_AUTHORITY_VALIDATION_STATUS]
+            ||[name isEqualToString:AD_TELEMETRY_EXTENDED_EXPIRES_ON_SETTING]
+            ||[name isEqualToString:AD_TELEMETRY_PROMPT_BEHAVIOR]
+            ||[name isEqualToString:AD_TELEMETRY_RESULT_STATUS]
+            ||[name isEqualToString:AD_TELEMETRY_IDP]
+            ||[name isEqualToString:AD_TELEMETRY_TENANT_ID]
+            ||[name isEqualToString:AD_TELEMETRY_USER_ID]
+            ||[name isEqualToString:AD_TELEMETRY_RESPONSE_TIME]
+            ||[name isEqualToString:AD_TELEMETRY_CLIENT_ID]
+            ||[name isEqualToString:AD_TELEMETRY_API_ID]
+            ||[name isEqualToString:AD_TELEMETRY_USER_CANCEL]
+            ||[name isEqualToString:AD_TELEMETRY_ERROR_CODE]
+            ||[name isEqualToString:AD_TELEMETRY_ERROR_DOMAIN]
+            ||[name isEqualToString:AD_TELEMETRY_PROTOCOL_CODE]
+            ||[name isEqualToString:AD_TELEMETRY_ERROR_DESCRIPTION])
         {
-            [eventToBeDispatched setObject:property.value forKey:property.name];
+            [eventToBeDispatched setObject:[properties objectForKey:name] forKey:name];
         }
     }
 }
 
+- (void)initAPIEventProperties
+{
+    [self setProperty:AD_TELEMETRY_RESULT_STATUS value:@""];
```

Comment:
It is said that each event not having all properties is a bug, such that one event cannot fully overwrite another event during integration. 

If we don't want it that way, I also have other solutions to make sure one event can fully overwrite another. But I am curious how other platforms solve this issue? :)